chringo/lib/main.dart

```diff
@@ -1,89 +1,116 @@
- @override
- Widget build(BuildContext context) {
-   return Scaffold(
-     appBar: AppBar(title: Text('Baby Name Votes')),
-     body: _buildBody(context),
-   );
- }
-
- Widget _buildBody(BuildContext context) {
-   // TODO: get actual snapshot from Cloud Firestore
-   return _buildList(context, dummySnapshot);
- }
-
- Widget _buildList(BuildContext context, List<Map> snapshot) {
-   return ListView(
-     padding: const EdgeInsets.only(top: 20.0),
-     children: snapshot.map((data) => _buildListItem(context, data)).toList(),
-   );
- }
-
- Widget _buildListItem(BuildContext context, Map data) {
-   final record = Record.fromMap(data);
-
-   return Padding(
-     key: ValueKey(record.name),
-     padding: const EdgeInsets.symmetric(horizontal: 16.0, vertical: 8.0),
-     child: Container(
-       decoration: BoxDecoration(
-         border: Border.all(color: Colors.grey),
-         borderRadius: BorderRadius.circular(5.0),
-       ),
-       child: ListTile(
-         title: Text(record.name),
-         trailing: Text(record.votes.toString()),
-         onTap: () => print(record),
-       ),
-     ),
-   );
- }
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(title: Text('CHRINGO')),
+      body: _buildBody(context),
+    );
+  }
+
+  Widget _buildBody(BuildContext context) {
+    return StreamBuilder<QuerySnapshot>(
+      stream: Firestore.instance.collection('wordbanks').snapshots(),
+      builder: (context, snapshot) {
+        Record wordsRecord;
+        if (snapshot.hasData) {
+          wordsRecord = Record.fromSnapshot(snapshot.data.documents.first);
+        }
+
+        return _buildGrid(context, wordsRecord.words);
+      },
+    );
+  }
+
+  Widget _buildGrid(BuildContext context, List<String> words) {
+    return GridView.count(
+      // Create a grid with 2 columns. If you change the scrollDirection to
+      // horizontal, this produces 2 rows.
+      crossAxisCount: 5,
+      // Generate 100 widgets that display their index in the List.
+      children: List.generate(25, (index) {
+        return new BingoCard(words[index]);
+      }),
+    );
+  }
+}
+
+class BingoCard extends StatefulWidget {
+  final String word;
+  BingoCard(this.word);
+
+  @override
+  _BingoCardState createState() => _BingoCardState(word);
+}
+
+class _BingoCardState extends State {
+  bool _isChecked = false;
+  final String word;
+  _BingoCardState(this.word);
+
+  @override
+  Widget build(BuildContext context) {
+    return GestureDetector(
+      child: Card(
+        color: _isChecked ? Colors.green : Colors.white,
+        child: Center(
+          child: Text(
+            word,
+            style: Theme.of(context).textTheme.subtitle,
+            textAlign: TextAlign.center,
+          ),
+        ),
+      ),
+      onTap: _toggleChecked,
+    );
+  }
+
+  void _toggleChecked() {
+    setState(() => _isChecked = !_isChecked);
+  }
 }
 
 class Record {
```

Comment:
It would also be good to move this class into a file in `records/wordbanks_collection.dart`